src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -538,4 +613,27 @@ export class HubConnection {
             type: MessageType.CancelInvocation,
         };
     }
+
+    private createStreamDataMessage(id: string, item: any): StreamDataMessage {
+        return {
+            item,
+            streamId: id,
+            type: MessageType.StreamData,
+        };
+    }
+
+    private createStreamCompleteMessage(id: string, error?: string): StreamCompleteMessage {
+        if (error) {
+            return {
+                error,
+                streamId: id,
+                type: MessageType.StreamComplete,
+            };
+        }
+
+        return {
+            streamId: id,
+            type: MessageType.StreamComplete,
```

Comment:
Because we're taking a random object `{}` and saying, hey this is now a `StreamCompleteMessage`, but it's still the same object underneath. Typescript is kind enough to complain when casting/returning the random object saying you don't have the required values set.